management/server/file_store.go

```diff
@@ -141,9 +136,52 @@ func (s *FileStore) AddAccount(account *Account) error {
 	return nil
 }
 
-// GetPeersForAPeer returns a list of peers available for a given peer (key)
+// GetAccountPeers returns a list of peers available for a given account
 // Effectively all the peers of the original peer's account if any
-func (s *FileStore) GetPeersForAPeer(peerKey string) ([]string, error) {
+func (s *FileStore) GetAccountPeers(accountId string) ([]*Peer, error) {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+
+	_, accountFound := s.Accounts[accountId]
+	if !accountFound {
+		return nil, status.Errorf(codes.Internal, "account not found")
+	}
+	var peers []*Peer
+	for _, peer := range s.Peers {
+		if peer.AccountId == accountId {
+			peers = append(peers, peer)
+		}
+	}
```

Comment:
I did it to make it a bit more explicit and clearly separate the Account entity from Peer in the store (so that they are handled separately). 
But I agree it looked better before, so I've changed it back and removed a couple of store methods